### DISCUSSION TITLE
BRouter avoidance settings

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#134] [Q&A] [answered]
We need to provide the BRouter profile options in Cruiser in a more user friendly way.

_Let's start by enabling **route avoidance settings** to work in BRouter (offline) routing:_
- Avoid motorways
- Avoid toll roads
- Avoid ferries
- Avoid unpaved roads
- ...

---

Offering more profile options is already planned and it is important that they all work **offline**!

**Comment by devemux86** [comment]
BRouter avoidance options are implemented in [Cruiser 3.0.18](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Available in "Settings | Routing | Options | Avoid" or search settings for "avoid".

BRouter avoidance options already work with [BRouter 1.6.3](https://play.google.com/store/apps/details?id=btools.routingapp) on Google Play.

<img src="https://user-images.githubusercontent.com/3484020/234191145-2330036a-b84a-46f3-baba-d197f2946f58.jpg" width="50%"/>

**Comment by DolomitesTom** [comment]
have to wait until Brouter is available via playstore

**Reply by devemux86** [reply]
@DolomitesTom 

BRouter avoidance options in Cruiser already work with [BRouter 1.6.3](https://play.google.com/store/apps/details?id=btools.routingapp) on Google Play. :slightly_smiling_face:

**Comment by MotoUKRider** [comment]
How absolute is the *Avoid Toll* setting in the app?

I am able to route, in Car profile, over the Humber Bridge when I enable this feature.  The OSM tags suggest that the `toll=yes` parameter is correct for both sides of the road.  In Cruiser app, it won't re-route the long way via a toll-free road but if I do the same journey on BRouter-web, I can toggle the *Avoid Toll" option and it does re-route as expected.

Does the app Car profile have a point where it ignores the setting to favour a much shorter route or is a parameter not being exchanged Cruiser<>BRouter?

PS: I recently took the plunge with OSM editing and added an exclusion on this bridge for motorcycles as these are free - which should make any *Avoid Toll" use in the motorcycle profile better with bridges/roads that allow free use.

**Reply by devemux86** [reply]
@MotoUKRider thanks, this is an interesting example.

You probably mean this bridge ([BRouter website route sample](https://brouter.de/brouter-web/#map=13/53.7081/-0.4437/standard&lonlats=-0.449796,53.694293;-0.451555,53.717332&profile=car-fast))?

Original BRouter profiles (like car-fast) have absolute avoid toll option (line 187).
Cruiser profile does not have absolute avoid options (to avoid failed routing).

You can test the Cruiser profile in BRouter website, profile + instructions here:
- https://github.com/devemux86/cruiser/discussions/272

We can make the avoid toll absolute by changing in line 260 the `10` to `0`:
`min if is_avoided_toll_road then 0 else 999`

Probably we need to offer absolute avoid options for some avoidances?
Or if possible to make it optional for the user.

**Reply by MotoUKRider** [reply]
Yes, that's the bridge.  It's a frustrating "feature" to have to either be aware of or work around.  I would expect the routing to follow the settings defined in the options - leaving the user to toggle each one in order to make the route more "routeable" :)  In that scenario, I feel message telling the user that their routing options/restrictions may be too "aggressive", or words to that effect?

In short, I have a problem with Cruiser *not* complying with my routing preference just because it *might* not be able to offer a solution.  An on/off switch should be just that (with my programmer's hat on!)

I've been watching the beta activity with interest, but haven't felt brave enough to try/break any of them yet - the forthcoming features do look good, though.

**Reply by devemux86** [reply]
Probably we need 3 levels for the BRouter avoidances?
- On (strict)
- On
- Off

**Reply by MotoUKRider** [reply]
Yes, that sounds like a fair solution.

**Reply by devemux86** [reply]
By the way, Kurviger seems to use this bridge even though avoid tolls is enabled (even on max strength).

BRouter in Cruiser with the new strict avoid options will not use this bridge.
(this already works on BRouter website with its default profiles)

**Comment by devemux86** [comment]
<img src="https://github.com/devemux86/cruiser/assets/3484020/7357af03-ee52-4b88-a409-ac2449a9ce49" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/e40eadbf-5741-42c6-aec5-57bfa31aabf9" width="50%"/>

**Comment by Leeb64** [comment]
Instead of 3 levels, 5 levels would be better.
At the middle level, the avoidance of primary roads is already too great. There are sometimes stupid detours...

**Reply by devemux86** [reply]
The penalties in BRouter profiles do not (?) seem to be linear,
so some testing is needed to decide levels for each road type.

**Comment by devemux86** [comment]
The updated driving-cruiser.brf profile is here:
- https://github.com/devemux86/cruiser/discussions/272

The avoid levels can be set in the various `_factor` parameters:
- 0: On (strong)
- 10: On

So you can test with other values in BRouter website
and we can see how to introduce more levels.

**Comment by MotoUKRider** [comment]
Thanks for the fix in the latest beta - the Car profile does indeed take the looooong way round!

But, despite updating BRouter just now, my motorcycle toll edit from 5 days ago on OSM doesn't seem to have propagated into the routing data - is that what I'm waiting for?  The motorcycle profile also re-routes when it *should* allow travel over the toll-free bridge.

On a related note, the automatic recalculation of a route when a critical setting is changed would be beneficial. I've noticed that the menu option for "Recalculate" on a route-line long-press is now 1-level deeper, and the only other way (that I've found) is to go into the Waypoints list, re-order something, move the item back to its original place and then tap "OK".

**Reply by devemux86** [reply]
BRouter data are updated daily.

Do you mean another toll?
Can you post the location and BRouter settings?

> the automatic recalculation of a route when a critical setting is changed would be beneficial

Probably via an option, as there are cases that we have ready or import a route,
then play with router settings just to modify them without wanting the route to change?

> the menu option for "Recalculate" on a route-line long-press is now 1-level deeper

The long press menu was restructured to avoid too many actions in 1st level.
We can always improve it and make changes based on everyone's daily use.
e.g. move "Recalculate route" on 1st level?

---

Another way to do a quick route recalculation also is:
- Long press on any waypoint and select move
- Do not move it, just press OK button :ballot_box_with_check:

**Reply by devemux86** [reply]
"Recalculate route" will be again in the 1st menu level in the next update.

**Reply by MotoUKRider** [reply]
It's the [Humber Bridge, northbound](https://www.openstreetmap.org/way/4063883) that I'm testing.

Confusingly, the [Dartford River Crossing](https://www.openstreetmap.org/way/20003982) routes a motorcycle the long way (to the Blackwall Tunnel, in London) rather than over the Dartford Bridge. Motorcycles are free, but using the new "strong" toll-avoid setting does not seem to take this into account.

I think Cruiser is avoiding a toll road based on the user avoidance setting *regardless* of the vehicle being allowed to pass "free".

**Reply by devemux86** [reply]
Trying to use the `toll:motorcycle` in BRouter profiles we get the message "unknown lookup name: toll:motorcycle".
So probably BRouter does not include the `toll:motorcycle` in its data.

Better to create an issue in [BRouter repository](https://github.com/abrensch/brouter/) and ask them about this tag?

cc @afischerdev

**Reply by MotoUKRider** [reply]
Interesting!  There is some satisfaction in knowing that I may have found an error/bug rather than doing something silly at my end.

I note that [this isn't the first time I've tried to be creative ](https://github.com/devemux86/cruiser/discussions/150) and found an issue :)

**Reply by devemux86** [reply]
@afischerdev added the `toll:motorcycle` in the future update of BRouter database.
- https://github.com/abrensch/brouter/pull/458

When the database update is merged, the new tags will come to our devices automatically
and then we can adapt the BRouter profiles with this tag.

@afischerdev many thanks!

**Reply by MotoUKRider** [reply]
@devemux86  This bridge again!  When I set "Avoid Toll Roads" to 100%, BRouter (internal) takes me on a long-winded detour yet this bridge is toll-free for motorcycles and the OSM tags that I edited some months back are still in place.

**Reply by devemux86** [reply]
@MotoUKRider 

BRouter has not yet included in its database the tag `toll:motorcycle`.
- https://github.com/abrensch/brouter/issues/639
- https://github.com/abrensch/brouter/pull/458

cc @afischerdev

**Reply by MotoUKRider** [reply]
Thanks - it's always comforting when it's not a user-generated issue!

**Comment by Leeb64** [comment]
So, I played a bit with the BRouter Web Client...
The following settings result in  useful motorcycle routes: (small value> big avoidance)
av.Toll>0,avMotorways>50,avPrimary>50,avFerrys>5,avUrban4(!!),avUnpaved,
valid for motorcycles,
consider Town, consider Forest.
Avoid Urban 4(1 is better!)  then "consider Forest" works without stupid shortcuts in towns.
Unfortunately the route calculation fails for routes over 200km(without shaping points)



[DRivingCRuiserXX.brf.txt.txt](https://github.com/devemux86/cruiser/files/13166031/DRivingCRuiserXX.brf.txt.txt)

**Reply by Leeb64** [reply]
For the avoidance setting, a slider with some kind of logarithmic scale would be suitable.
0= complete avoidance(not so good!),
1=strong avoidance,
10-30 = moderate avoidance, 60 = weak avoidance.
The slider from 0-60, with 10 in the middle (or something similar,Between "0" and "1" a small space).
Avoidance completely switched off -> ~1000.

When using "consider forest", the "avoid urban" factor should be set to 1, otherwise strange routes are often calculated within built-up areas.
"0" should be avoided!
(But 0 can be useful, for example, when avoiding tolls)

The BRouter also takes quite a long time to calculate.
Wouldn't it be possible to leave out this kinematics stuff somehow?
Nobody actually cares about that when riding a motorcycle...

**Reply by devemux86** [reply]
@Leeb64 thanks for the additional info and the tests!

> 0= complete avoidance(not so good!)

That's why we didn't have it until now, it could make BRouter failing to find routes if many avoidances are set to 0.

Regarding the slider, unfortunately Android does not provide such UI conveniences,
I will have to implement it or we need to think for another selection method.

So we seem to decide for avoidance levels:
- 0: complete
- 1: strong
- ...: middle
- 60: weak

These levels are for all avoidance road types in the main options screen?

>  Wouldn't it be possible to leave out this kinematics stuff somehow?

Why do you think the kinematics is the reason?
Or did you find @poutnikl's profiles faster?

Too many avoidances are often the reason, especially the new ones in the "Other" category.
These are still work in progress and they may need improved integration in Cruiser's profile.

Also see this post by @EssBee59 one of BRouter's developers:
- https://github.com/devemux86/cruiser/discussions/266#discussioncomment-7295972

cc @afischerdev @EssBee59

**Reply by Leeb64** [reply]
The kinematics thing is just a guess...
The @poutnikl profiles are much faster!

**Reply by Leeb64** [reply]
These avoidance settings should be adjustable for all street types (including residential streets).
From factor 1! up to about 60(adjustable) and completely off (1000 or so).

**Reply by devemux86** [reply]
We could use a similar interface with levels using factors internally that we can improve / adjust anytime:
- Off
- Strength 1 or Weak (factor 60)
- Strength 2 or Low (factor ?)
- Strength 3 or High (factor ?)
- Strength 4 or Strong (factor 1)
- Strength 5 or Strict (factor 0)

**Reply by umpfmpf** [reply]
That would be the same like in Kurviger. 
Sounds good.

**Reply by Leeb64** [reply]
Something like this...
Strength 2: ~ 22
Strength 3: ~  7
That could work.
It always depends a bit on the road conditions.

**Reply by devemux86** [reply]
The benefit of this solution is that the factors are internal and we can improve them anytime.

We can apply these levels to all BRouter avoidances:
[Toll roads, Ferries, Motorways, Primary roads, Secondary roads, Tertiary roads, Residential roads]

@Leeb64 Do you think we can use the same "middle" values for all avoidances?

**Comment by EssBee59** [comment]
Hello,
As I get too much mails I will now unsubscribe from this discussion.
I think,  basically yes, different values are better as only white or black.
My last remark: it seems that currently the list of the parameters is hard coded (it do not change when I select a different profile)

In the same way as the brouter-web, the parameter-list should be created dynamically according to the profile itself (parameter having % in the comment such as ==> #  %consider_traffic% | )
Regards

**Reply by devemux86** [reply]
@EssBee59

Do you mean when users open an external BRouter profile?
This is more complicated. And it could only work in the 1 language of the profile texts.

The predefined profiles like driving-cruiser and BRouter original (bike, hike) used in Cruiser
already display different settings when users select these internal profiles, see images below.

<table>
<tr>
<th>Driving</th><th>Cycling</th><th>Walking</th>
</tr>
<tr>
<td><img src="https://github.com/devemux86/cruiser/assets/3484020/6f087975-6c39-4d36-a054-f833796072f3"/></td><td><img src="https://github.com/devemux86/cruiser/assets/3484020/9525d901-c71f-4841-8200-92a747eea3c6"/></td><td><img src="https://github.com/devemux86/cruiser/assets/3484020/84c1307f-3afb-41eb-8a4c-21f80628ea6e"/></td>
</tr>
</table>

**Comment by Leeb64** [comment]
I played with the values ​​again...
There's a lot going on at the bottom, but not at the top.

Avoidance levels that work reasonably well would be (for now)
1,11,22,33,44.
Important: when using Consider Town,Consider Forest,Consider Traffic,ConsiderNoise, ->set the value for "avoid urban" to 1 and the value for "avoid terretory" max to 11!
Otherwise things will get weird in cities!
Perhaps when you select these criteria, this default should be done automatically.

I can create  useful motorcycle routes with these settings:
Avoid Toll 1,Avoid Motorways 11,Avoid Ferrys 1,Avoid Primary11, Avoid Secondary->None,
Avoid Teritary11,Avoid Urban1,+Valid for Motorcycles,+Consider Town,+Consider Forest

**Reply by devemux86** [reply]
@Leeb64 thanks for the tests!

By the way, if we look carefully where the avoid values are applied in the profile code,
it is in the "assign maxspeed" block, so larger values are expected to negate the avoidances.

**Reply by devemux86** [reply]
>  1,11,22,33,44.

This together with "0" are 6 levels, we need 5 levels or a number of levels that can easily divide the 100.
Users do not need to be confused with internal detailed factors, just a generic tool to adjust the strengths.

So we can have a simple % slider with values:
- 0% (Off)
- 20% factor ?
- 40% factor ?
- 60% factor ?
- 80% factor 1
- 100% factor 0

**Reply by Leeb64** [reply]
Very simplified... the whole thing depends entirely on the roads available in the respective area...
60%>>10
40%>>22
20%>>45

As already written above, the avoidance values ​​Urban=1 and Tertiary=(max)11 should be adhered to when using Cosider Town,Traffic,Noise,Forest,River.

there's still the speed problem...
What could be done to speed up the calculation?

**Reply by devemux86** [reply]
> What could be done to speed up the calculation?

Use only the avoidances in the main screen (not in the "Other" screen).

I do not know how the new "consider" parameters work or if they are defined correctly in Cruiser profile.
The avoidances in the main screen should be fine for the most cases.
Then the speed depends on the number of avoidances and the device cpu.

**Reply by devemux86** [reply]
For better speed:
- Use many waypoints (as BRouter calculates each segment separately)
- Do not enable alternative routes (as they multiply the calculation time)

**Reply by 0709wiwiwi** [reply]
I may be throwing a cudgel into a pigeonhole but aren't we on an aberration by using that slowish walk and bicycle offline Brouter ? 

So you ask a (car) GPS router to calculate an efficient route to a target 300km away. If you then have to determine the middle shaping points yourself, you're better off driving back using your paper street atlas.

I did next exercise using the Guru maps free version (offline) and a single target point 300 km away and avoid highways plus avoid tollroads set. (Efficiency is the goal not kurvig). This calculation (with 3 alternatives to choose from) is done and ready within 2 sec. In navigation when you trigger an auto recalculation to this single target 300 km away the rerouting result follows as equally fast.

**Reply by devemux86** [reply]
BRouter is not just for bikes, despite its name, that's a misconception!
It can even calculate routes for rivers, trains, boats, all travel modes.

And obviously for motor road vehicles, like cars, motorcycles, moped, etc.

As long as there is a routable network in OpenStreetMap, it can be used in a routing engine (if it is flexible enough to parse it).

---

Regarding the speed, it is a cost for the increased flexibility.
The more options, the more avoidances, the less available roads to use and more time to find a result.

If you have specific examples it is better to also inform BRouter developers in [BRouter repository](https://github.com/abrensch/brouter/).
Especially if we compare BRouter's website server and an offline device's cpu, there is always a trade-off.

**Reply by 0709wiwiwi** [reply]
> It can even calculate routes for rivers, trains, boats, all travel modes.

Anyway the Brouter is becoming agonisingly slow at the longer distances.
https://github.com/abrensch/brouter/issues/436#issuecomment-1172483456
Moreover, the offline performance should be tested with the Android phone(s) used.

Traject test: Beeline line distance start to end: 249.5 km.
- Brouter: Avoid motorways: 20%.
- Guru router: Avoid motorways.
Calculation times. 
Mi A2 Lite 
- Brouter: 70 seconds. (Single)
- Guru: 2 seconds (3 alternatives).
Motorola G42: 
- Brouter: 28 seconds. (Single)
- Guru: < 2 secs (3 alternatives shown)

**Reply by devemux86** [reply]
What routing options and flexibility does the other router offers?

An answer about speed vs flexibility could be this one:
- https://github.com/abrensch/brouter/issues/436#issuecomment-1172500187

**Reply by 0709wiwiwi** [reply]
The topic is not about missing flexibility as we sure agree that the shortlink to the devs is very valuable.
What is succesfully tested online must additionally be confirmed by real android phones offline operation.
I notice and so do report that for expected performant long distance operations the Brouter is  slow.  :((
Facts and I have double checked by 2 phones.

**Reply by poutnikl** [reply]
Note the BRouter uses as the routing algorithm the modified  A-star algorithm, similarly as OsmAnd, that is optimized for bicycle/hiking routing, or other high-resolution routing, potentially considering all possible routing related Openstreetmaps data. The route calculation time generally raises with the squared bee-line distance of the route segment endpoints.

Car routing apps (GH included) usually use Contracted Hierarchy (CH) algorithm or similar, that uses **highly** preprocessed routing data, mostly on the server side with the exception of the local  GH offline data. They get dynamically simplified when the large scale is evaluated for the route. These are optimized for low resolution long distance routes like those done by cars, with very limited options to modify the routing, in high contrast to highly customizable BRouter routing.

Comparing speed performance of A-star based and CH based routing for car routes is like comparing speed of bicycles and cars for a planned 500 km long trip. Everybody knows cars are faster than bicycles, no sense in complaining about that. Choosing the bicycle for that is the free choice, but do not expect to be there in 5 hours.

**Reply by devemux86** [reply]
@poutnikl

May I ask how the performance of BRouter kinematic model and your profiles compare?
https://github.com/poutnikl/Brouter-profiles/tree/master/CarProfiles

I had modified them in the past and managed to include similar detailed avoidances.

**Reply by 0709wiwiwi** [reply]
Hi Libor. 
I am not complaining about this, but it is still better to mention it indeed. There are a lot of efforts being made here to get a desired curvy route. But those extra rules also add to the reaction slowness. Opinions may differ but with an autorecalculation trigger it does have the consequence (car gps) that the calculation barely stops and I find that very annoying.  At cycling speeds and the usual shorter trajects the autorecalculation by the BRouter now works excellently. (+/- based on your original idea and proposal in that other forum, by the way, Libor)

**Reply by devemux86** [reply]
Rerouting in Cruiser calculates only a route until the next waypoint, never the whole existing route again, so it should be fast.

Naturally we should have some waypoints in our route, cannot expect instant rerouting with distance of start - end = 250 km.
(and several avoidances)

**Reply by poutnikl** [reply]
@devemux86 I am not sure, as I have not used my profiles for few years and the comparison is similarly old. 
I developed them as a workaround, using few less or more dirty tricks, as there was no good car profile until abrensch developed the kinematic model. 

I remember using my profiles in bidirectional long distance routing from Brno (Czechia) to Leon (Spain) with 150-200 km long route segments (using viapoints or shaping points). BRouter is unbeaten if map details are essential.

The routing algorithm is the same for my profiles and kinematic model, but the cost (equivalent length) of road segments are calculated differently. Explicitly by expressions for costfactor in my profiles versus internally by kinematic models. I guess the latter was somewhat faster.

**Reply by Leeb64** [reply]
(my)Route calculation time:(with these motorcycle settings)
300km->75s
200km->35s,
100km->8s,
75km->5s
50km->2s,
A waypoint about every 50-80km then that shouldn't be a problem.
The waypoints (or shaping points) should be set anyway so that if you have to leave the planned route, you can end up back on it at some point.

**Reply by 0709wiwiwi** [reply]
Just is an everyday task for a car or motorbike GPS.
Navigate efficiently to a target +/- 200 km away and avoid motorways.  You occasionally will sure trigger the (auto)recalculation on this traject. This process should be smooth and fast. How smoothly can it be done in OFFLINE mode ?  Motorbiker 1 uses the Guru_router and drives ahead. Biker 2 follows. See the video and compare the results. [Autorecalculation test](https://youtu.be/etOShrdodIIurl)

**Reply by Leeb64** [reply]
This is not about the most effective route guidance possible, but rather about the most curvy (ineffective?) route guidance possible.
Motorcyclists know what I mean.->all Just(more or less) for fun!

**Reply by 0709wiwiwi** [reply]
I do know. But there are also occasions when you just want to get somewhere as efficiently as possible right ? The Cruiser app announces it is not only "Kurviger" but has something more and extra to offer I thought right ? By the way, you don't drive around as swerving as possible every time you leave your house, do you ? At least where I live, you won't have that much fun ;-)  Motorcyclists (that I know of) here ride further (efficiently) towards the Ardennes or the Eifel, drive to a hotel and spend a weekend kurvig there. Besides, as a cyclist, I am quite happy with the Brouter. I think I also made it clear that I tested this same BRouter as a car gps over plenty of distance. And for that job it is not optimised and therefore not the best choice. That is nothing more than an observation.  Putting together a somewhat longer (kurvig) route with a router that is quite slow to respond doesn't seem very appealing to me either, by the way. But I'm me, no problem with the well-liked.

**Reply by Leeb64** [reply]
That's the good thing about the cruiser:
You can choose the appropriate profile depending on your needs.

**Reply by 0709wiwiwi** [reply]
@Leeb64  @devemux86 
Sure I would prefer designs using Brouter, (free) but how best to deal with that inertia ?
Generating a Brouter design to a single long-range target takes a long time. That is a fact.
Anyway a first and completely new idea (after breakfast) could possibly help a little  ?
If you then want to fine tune later with a first initial Shaping point it still does takes a long reaction time.
What I imagine is this.  (Please think with me...is really a first idea)
Cruiser should automatically place a normal Shaping point every 50 km already.
You can then manipulate these shaping points as you wish with a good workable response time.
Recalculating only the 50 km traject before and 50km after the manipulated shaping point. 
This concerns so 100km to be recalculated. The BRouter is so able to react comfortably fast.

**Reply by poutnikl** [reply]
@0709wiwiwi You idea seems being similar to what I have often done for long, manually planned routes with many initial manual shaping/via points, using this repeated sequence:

- removing one of the shaping points
- letting BR calculate the route segment between its neighbor shaping points
- Placing the new shaping point near the middle of the newly calculated route segment.
- Not touching eventual viapoints, as they are there for reason.

This way the series of shaping points are kind of both manually selected and calculated by BRouter at the same time.

**Comment by devemux86** [comment]
For example something like this.
With red the absolute 100% avoidance (factor 0), that could lead to no routing.

<img src="https://github.com/devemux86/cruiser/assets/3484020/4854d63c-175c-488a-aebc-5c346b6d7412" width="50%"/>

**Reply by devemux86** [reply]
The new BRouter avoid levels are available in the latest:
- https://github.com/devemux86/cruiser/discussions/329

---

- 0% (Off)
- 20% factor 45
- 40% factor 22
- 60% factor 10
- 80% factor 1
- 100% factor 0 (absolute)

**Reply by Leeb64** [reply]
A little playing around with the settings and you have a route that rivals “Kurviger”...
Settings:
>Toll 80%>,Ferry>80%,
Highway>80%, main road>0%, secondary road>0%, tertiary road>40%, residential road>100%.
Consider forest, consider town

Much slower...but Offline!

**Reply by umpfmpf** [reply]
Yes! 
This is the first time that the route drives all the winding roads.  And runs almost the same route as "Kurviger". 
The only thing missing is the avoidance of driving the road twice.  Otherwise: great work

**Reply by devemux86** [reply]
@umpfmpf can you also share what BRouter options do you usually use?

**Reply by umpfmpf** [reply]
In the old settings: toll on, ferries on, motorway off, other roads on, residential areas off, unpaved roads out, gravel off. 
Then avoid cities and set the speed to 94 km / h. 
With the new setting options, I have to check first. 
The proposed ones from "Leeb64" look very good. 
Anyway better than ever!

**Reply by devemux86** [reply]
> set the speed to 94 km / h

BRouter kinematic model offers this extra, we can avoid roads **also** with the maximum speed option.
(combined with avoid options or not)

**Reply by umpfmpf** [reply]
I tried the route planning at different speeds until I found the optimal route guidance. Now with the new possibilities of setting, I have to try this again. In the old version it made a huge difference in what speed I set.

**Reply by Leeb64** [reply]
I don't really understand the logic of the set speed...(When the speed is set higher, "smaller" roads are sometimes taken...)
But it definitely makes a difference.
Everyone has to decide for themselves.
It's also very practical. You can vary it.

**Reply by umpfmpf** [reply]
And I also don't understand the logic of the weight.

**Reply by devemux86** [reply]
Both "Maximum speed" and "Total weight" are not used directly inside the profile rules,
but they are used in [BRouter code](https://github.com/abrensch/brouter/blob/master/brouter-core/src/main/java/btools/router/KinematicModel.java).

**Reply by Ciutnik** [reply]
I can confirm that the new avoidances are excellent! Great job, thank you very much!

**Reply by devemux86** [reply]
Thanks also to @Leeb64 for the tests and the avoidance strengths (factors)! :+1:

**Reply by Leeb64** [reply]
The route calculation now works quite well.
For longer routes, depending on the avoidance settings, the BRouter stops the calculation after a certain time.
You could simply split the route now.
For those who are too lazy for that(me)....would there perhaps be an option to move/double the route calculation time limit of the BRouter?

**Reply by devemux86** [reply]
@Leeb64 What time? Any message?

The routing timeout is currently set to 10 min. It seems too long to wait for a result.
Use more waypoints usually helps. Also disable alternative routes.

**Reply by Leeb64** [reply]
10 minutes?
For me, the BRouter stops calculating after about 40 seconds!(ca250km)
Report:
Route not found: Add or move waypoints

**Reply by devemux86** [reply]
@Leeb64 Please provide the route and BRouter options so we can test it.
You can calculate it with another router and export it as json.

>  Route not found: Add or move waypoints

So this not a timeout, but cannot calculate a route with the selected waypoints and avoid options.

**Reply by devemux86** [reply]
Or could be that the selected avoidance options cannot calculate a route.
e.g. set a waypoint on a primary road and also forbid primary roads

It is often best to use a few avoidances and increase them gradually.

**Reply by Leeb64** [reply]
Route options:
Toll 100%
Ferries100%Highways80%Main roads0%Secondary roads0%Tertiary roads40%
residential streets 100% 
forests + green regions
Avoid cities
Speed132km
Weight 310kg.
With these settings it works.Calculation time approx. 35s.
If I change the avoidance of residential streets to 80%, it doesn't work.More than 40s.

[Test.json](https://github.com/devemux86/cruiser/files/13196095/Test.json)

**Reply by Leeb64** [reply]
..It works with my other smartphone (Doogee)!!
This takes about 1 minute for the whole calculation.

**Reply by Leeb64** [reply]
Reinstalled BRouter -> no difference

**Reply by devemux86** [reply]
@Leeb64 thanks for the route.
It works here, it probably depends on the device.

About the new options for cities, forests, etc. better use them with caution, they are still under development.
Better set some waypoints to avoid a city or enter a forest. It will also improve greatly the calculation speed!

In the future we will design no-go areas in route planning, so it will be easier to avoid / prefer certain regions.
(like in BRouter website)

Also better to not change the max speed and total weight from the defaults if want to use many avoidances.
Kinematic model options and too many avoidances at the same time and with only 2 waypoints can be too much work for some cpu.

**Reply by devemux86** [reply]
Decreasing the "Maximum speed" or "Total weight" reduces the available road network in route calculation.
So we can imagine that these combined with all the avoidance options how fewer roads are left to find a route.

The same applies to the cities / forests / ... options. They also strain the engine and limit the available roads.

A recommendation would be to stick with the avoid options on the main screen.
And possibly use "Maximum speed" to micromanage on top.

(at least until we understand / improve better the other options) :slightly_smiling_face:

**Reply by Leeb64** [reply]
The Sony (it doesn't work with that) is much faster than the Doogee.
Antutu Sony~350000,
Doogee~80000.
It also calculates shorter routes faster than the Doogee.
Strange..

**Reply by devemux86** [reply]
I remember an old recommendation to add a waypoint every 100 km?
Or is it another distance?

Also instead of starting with a long route, create it step by step, adding extra segments via waypoints.

For BRouter this is more relevant because of its multi-segment calculation algorithm.

**Reply by Leeb64** [reply]
Stupid mistake...
I installed the BRouter program again - but without renewing the segments.
Now I have renewed the segments...now it works.

**Comment by Leeb64** [comment]
Until you find the optimal settings for the "curvier motorcycle profile", some people may lose interest...
If you had a preconfigured profile (which can still be changed), that might be an advantage.
Good default settings would be:
Toll 80%(0-100), Ferries 80%(40-100), Motorway 60%(0-100), Main roads 0%(0-80), Secondary roads 0%(0-60), Tertiary roads 60%(20-80), Residential roads 80%(80-100),
prefer forests and green regions,
Avoid cities
Vmax 130km(50-250),
Weight 1500kg.
When you call up this profile, it is preset, but can be changed as needed

**Reply by devemux86** [reply]
@Leeb64 this is a good idea.

I had asked in another topic if we could include presets of options for easier use:
- https://github.com/devemux86/cruiser/discussions/254

**Comment by devemux86** [comment, marked as answer]
BRouter (offline): configure the strength of avoidances is available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by Leeb64** [comment]
As avoidance categories we have:
Highways,Main roads,Secondary roads,Tertiary roads,"(????)",Residential roads,Unpaved roads....
A category "below" the tertiary roads would be: Highway unclassified->(????)
Could this category be added?

**Reply by devemux86** [reply]
@Leeb64 can you post the OpenStreetMap tag(s)?

Each avoidance category works with specific OpenStreetMap tags.
(they can be seen inside the profile)

**Reply by Leeb64** [reply]
![grafik](https://github.com/devemux86/cruiser/assets/120722053/a0d57ee2-d8c4-4fdd-a86a-0fbb48700286)
As you can see in the picture, "shortcuts" are taken that you don't really want...
Teritary is good!
unclassified is usually not that good!

**Reply by devemux86** [reply]
"Tertiary" category currently contains highway=tertiary|tertiary_link and [highway=unclassified](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dunclassified).

**Reply by Leeb64** [reply]
That's what I thought.
Especially for motorcycle tours, it would be helpful to avoid these unclassified roads. Tertiary roads are usually quite nice!

**Reply by Leeb64** [reply]
Here is a trick:
![grafik](https://github.com/devemux86/cruiser/assets/120722053/608c16b7-5491-4645-b0a5-cc72afdc0f5d)
Reduce the speed of highway unclassified!(original 50km/h)

**Reply by Leeb64** [reply]
Here a Motocycle-Profil to test:
[sam1500 - Kopie.brf.txt](https://github.com/devemux86/cruiser/files/14623871/sam1500.-.Kopie.brf.txt)
Characteristic:
No tolls, almost no motorway(bypassing towns), few main roads, lots of secondary roads, more tertiary roads, few unclassified roads.Avoid citys.

**Reply by devemux86** [reply]
> A category "below" the tertiary roads would be: Highway unclassified->(????)
Could this category be added?

@Leeb64 an updated driving-cruiser profile with separate "unclassified" avoidance is here:

- https://github.com/devemux86/cruiser/discussions/272

**Reply by Leeb64** [reply]
Very good! That brings a lot!
Avoid unclassifed 5
Avoid urban 3
Avoid Primary 15
Avoid  Motorways 10
Avoid Toll 1
Avoid Ferrys 0
the Rest :not Avoid
Consider Town
Consider Forest
First Test->Works fine
